из api импорта PetFriends
из настроек импорт valid_email, valid_password
импорт OS


pf = Друзья для домашних животных()


def test_get_api_key_for_valid_user(email=valid_email, password=valid_password):
    статус, результат = pf. get_api_key(email, пароль))
     утвердить статус == 200
    утверждение 'key' в результате

def test_get_all_pets_with_valid_key(filter=''):
    _, auth_key = pf. get_api_key(valid_email, valid_password)
    статус, результат = pf. get_list_of_pets(auth_key, фильтр)
     утвердить статус == 200
    assert len(result['pets']) > 0

def test_add_new_pet_simple(name='Kotya', animal_type='Catt', age='1'):
    _, auth_key = pf. get_api_key(valid_email, valid_password)
    статус, результат = pf. add_new_pet_simple(auth_key, имя, animal_type, возраст)
     утвердить статус == 200
    assert result['name'] == name

def test_add_new_pet_with_valid_data(name='Kotya', animal_type='Catt', age='1', pet_photo='images/Kotya.jpeg'):
    pet_photo = ос. путь. join(os. путь. имя(__file__), pet_photo)
    _, auth_key = pf. get_api_key(valid_email, valid_password)
    
    статус, результат = pf. add_new_pet(auth_key, имя, animal_type, возраст, pet_photo)
     утвердить статус == 200
    assert result['name'] == name
    
def test_successful_delete_self_pet():
    _, auth_key = pf. get_api_key(valid_email, valid_password)
    _, my_pets = pf. get_list_of_pets(auth_key, "my_pets")
    если len(my_pets['pets']) == 0:
        пф. add_new_pet(auth_key, "Супец", "ммм", "2", "изображения/котя.jpeg")
        _, my_pets = pf. get_list_of_pets(auth_key, "my_pets")

    pet_id = my_pets['pets'][0]['id']
    статус, _ = pf. delete_pet(auth_key, pet_id)

    _, my_pets = pf. get_list_of_pets(auth_key, "my_pets")

     утвердить статус == 200
    утверждают pet_id не в my_pets. Значения()

def test_successful_update_self_pet_info(name='Murz', animal_type='CatDog', age=3):
    _, auth_key = pf. get_api_key(valid_email, valid_password)
    _, my_pets = pf. get_list_of_pets(auth_key, "my_pets")

    если len(my_pets['pets']) > 0:
        статус, результат = pf. update_pet_info(auth_key, my_pets['pets'][0]['id'], имя, animal_type, возраст)

         утвердить статус == 200
        assert result['name'] == name
    еще:
        raise Exception('Пустой список my_pets')

def test_add_new_pet_photo(pet_photo='images/Kotya.jpeg'):
    pet_photo = ос. путь. join(os. путь. имя(__file__), pet_photo)
    _, auth_key = pf. get_api_key(valid_email, valid_password)
    _, my_pets = pf. get_list_of_pets(auth_key, "my_pets")
    
    если len(my_pets['pets']) > 0:
        pet_id = my_pets['pets'][0]['id']
        status, result = pf.add_pet_photo(auth_key, pet_id, pet_photo)

    assert status == 200
    assert 'pet_photo' in result

#Negative tests below

def test_get_api_key_for_invalid_user(email='invalid@mmm.ru', password=valid_password):
    status, result = pf.get_api_key(email, password)
    assert status == 403
    assert 'Forbidden' in result
# Negative test below
def test_get_api_key_for_empty_user(email='', password=valid_password):
    status, result = pf.get_api_key(email, password)
    assert status == 403
    assert 'Forbidden' in result
# Negative test below
def test_get_api_key_for_invalid_pswrd(email=valid_email, password='678'):
    status, result = pf.get_api_key(email, password)
    assert status == 403
    assert 'Forbidden' in result
# Negative test below
def test_get_api_key_for_empty_pswrd(email=valid_email, password=''):
    status, result = pf.get_api_key(email, password)
    assert status == 403
    assert 'Forbidden' in result
# Negative test below
def test_get_api_key_for_incorrect_userdata(email='23', password='password'):
    status, result = pf.get_api_key(email, password)
    assert status == 403
    assert 'Forbidden' in result
# Negative test below
def test_get_all_pets_with_outdated_auth_key(filter=''):
    auth_key = {'key': 'd3c1355cc3c551acbebe5b58ead5d09897aa6ff03ce448554d40987d'}
    status, result = pf.get_list_of_pets(auth_key, filter)
    assert status == 403
    assert 'Forbidden' in result
# Negative test below
def test_get_all_pets_with_symbols_in_auth_key(filter=''):
    auth_key = {'key': '!@#$%^&*!@#$%^&*!@#$%^&*!@#$%^&!@#$%^&*!@#$%^&*!@#$%^&*@'}
    status, result = pf.get_list_of_pets(auth_key, filter)
    assert status == 403
    assert 'Forbidden' in result
# Negative test below
def test_get_all_pets_with_incorrect_filter(filter='pets'):
    _, auth_key = pf.get_api_key(valid_email, valid_password)
    status, result = pf.get_list_of_pets(auth_key, filter)
     утвердить статус == 500
    Утверждение 'Внутренняя ошибка сервера' в результате
# Отрицательный тест ниже
def test_get_my_pets_with_cyrillic_in_filter(filter='ьн_зуеы'):
    _, auth_key = pf. get_api_key(valid_email, valid_password)
    статус, результат = pf. get_list_of_pets(auth_key, фильтр)
     утвердить статус == 500
    Утверждение 'Внутренняя ошибка сервера' в результате
# Отрицательный тест ниже
def test_get_all_pets_with_255_characters_in_filter(filter='123456789123456789123456789123456789123456789123456789123456789123456789123123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789123456789'):
    _, auth_key = pf. get_api_key(valid_email, valid_password)
    статус, результат = pf. get_list_of_pets(auth_key, фильтр)
     утвердить статус == 500
    Утверждение 'Внутренняя ошибка сервера' в результате
# Отрицательный тест ниже
def test_unsuccessful_delete_self_pet_with_empty_pet_id():
    _, auth_key = pf. get_api_key(valid_email, valid_password)
    _, my_pets = pf. get_list_of_pets(auth_key, "my_pets")
    если len(my_pets['pets']) == 0:
        пф. add_new_pet(auth_key, "Супец", "ммм", "2", "изображения/котя.jpeg")
        _, my_pets = pf. get_list_of_pets(auth_key, "my_pets")

    pet_id = ''
    статус, _ = pf. delete_pet(auth_key, pet_id)

    _, my_pets = pf. get_list_of_pets(auth_key, "my_pets")

     assert status == 404
    утверждают pet_id не в my_pets. Значения()
